stdlib/public/Darwin/Accelerate/vImage_Buffer.swift

```diff
@@ -0,0 +1,348 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Accelerate
+
+//===----------------------------------------------------------------------===//
+//
+//  vImage_Buffer
+//
+//===----------------------------------------------------------------------===//
+
+@available(iOS 9999, OSX 9999, tvOS 9999, watchOS 9999, *)
+extension vImage_Buffer {
+    
+    /// Returns the size, in pixels, of a vImage buffer.
+    public var size: CGSize {
+        var mutableSelf = self
+        return vImageBuffer_GetSize(&mutableSelf)
+    }
+    
+    //===----------------------------------------------------------------------===//
+    
+    /// Initializes a vImage buffer of a specified size without memory allocation,
+    /// reporting the preferred alignment.
+    ///
+    /// - Parameter size: The size of the image.
+    /// - Parameter bitsPerPixel: The number of bits in a pixel of image data.
+    /// - Parameter alignment: The preferred alignment is written to this parameter.
+    ///
+    /// - Returns: An initialized vImage buffer.
+    public init?(size: CGSize,
```

Comment:
It's *really* not OK to have an `init` function that leaves `self` in an uninitialized state. This needs re-working. What pattern are you trying to enable by providing this?

Reply:
But it does have `self.init()` (I've just moved it to after the size check). This initializer initializes a buffer without memory allocation (i.e. with `kvImageNoAllocate`) and writes the alignment to the `alignment` inout parameter.

Reply:
The problem is that after calling this, the resulting vImage_Buffer object still isn't initialized. If the goal is to get the desired rowBytes and alignment, let's have a static function that *just* does that, rather than lying about it being an initializer.

Reply:
Good point. I've removed that faux initializer and replaced with a new static function: `preferredAlignmentAndRowBytes`.